src/ProjectTemplates/test/ByteOrderMarkTest.cs

```diff
@@ -58,5 +58,32 @@ public void CheckForByteOrderMark_ForAllTemplates(string projectName)
 
             Assert.False(filesWithBOMCharactersPresent);
         }
+
+        [Fact]
+        public void RazorFilesInWebProjects_ShouldContainBOM()
```

Comment:
Good test to have. Is there some context on why Razor should have BOM?

Reply:
> Is there some context on why Razor should have BOM?

Unfortunately I don't have any. @mkArtakMSFT and I spoke to the Web Tools team and they said they don't have any hard requirement on anything and VS respects the encoding of the original checked-in templates. My goal for this PR was to keep everything consistent once way or the other. I noticed most of the Razor files are with BOM and decided to make them all that way.